LiteCore/tests/VectorQueryTest.cc

```diff
@@ -227,23 +227,160 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid Vector Query", "[Query][.Vec
+        } else {
+            // Assert that the SQL++ query yields exactly same result as the JSON query.
+            auto result0 = collectResults(query->createEnumerator(&options));
+            CHECK(result0 == result);
+        }
+    }
+}
 
-    Log("---- Querying with $target = data");
-    Query::Options options = optionsWithTargetVector(kTargetVector, kData);
-    checkExpectedResults(query->createEnumerator(&options),
-                         {"rec-5300", "rec-4900", "rec-7100", "rec-3600", "rec-8700", "rec-8500", "rec-2400",
-                          "rec-4700", "rec-4300", "rec-2600"},
-                         {85776, 90431, 92142, 92629, 94598, 94989, 104787, 106750, 113260, 116129});
+N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid vs plain Vector Query", "[Query][.VectorSearch]") {
+    readVectorDocs();
+    {
+        // Add some docs without vector data, to ensure that doesn't break indexing:
+        ExclusiveTransaction t(db);
+        writeMultipleTypeDocs(t);
+        t.commit();
+    }
+    createVectorIndex();
+
+    // Case 1: plain query because there is no additional term in the WHERE clause
+    // ===================
+
+    string queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+                      + " WHERE VECTOR_MATCH(vecIndex, $target) "
+                        " ORDER BY distance LIMIT 10";
+    Retained<Query> query{store->compileQuery(queryStr, QueryLanguage::kN1QL)};
+
+    Query::Options                        options = optionsWithTargetVector(kTargetVector, kData);
+    std::vector<std::pair<string, float>> result  = collectResults(query->createEnumerator(&options));
+
+    // The outer limit 10 is provided to VECTOR_MATCH as argument max_results
+    CHECK(result.size() == 10);
+
+
+    // Case 2: plain query because of explicitly assigned max_results to VECTOR_MATCH
+    // ===================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target, 10) "
+                 " ORDER BY distance";
+    query                                         = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    std::vector<std::pair<string, float>> result0 = collectResults(query->createEnumerator(&options));
+
+    // Same effect as test 1.
+    CHECK(result0 == result);
+
+
+    // Case3: plain query. If not specified, VECTOR_MATCH will use 3 as max_results.
+    // ==================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target) "
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    // Argument max_results to VECTOR_MATCH is set by defaut to 3.
```

Comment:
Do we need to test that we get exactly 3 results? It's not an important number, just a placeholder default.

Reply:
Otherwise, how do we that default 3 is used? Or, I can put the test out to some where else we can examine the final sql for sqlite. Putting it here helps to understand the matter better. By the way, it's also testing that hybrid logic is used here.